crates/solver/src/settlement_ranker.rs

```diff
@@ -159,12 +162,15 @@ impl SettlementRanker {
             rated_settlements.len(),
             errors.len(),
         );
-        for (solver, settlement, _, error) in &errors {
+        for (solver, settlement, _, block_number, error) in &errors {
             solver.notify_auction_result(
                 auction_id,
                 AuctionResult::Rejected(SolverRejectionReason::SimulationFailure(
-                    error.to_string(),
-                    call_data(settlement.clone().into()),
+                    SimulationFailureParams {
```

Comment:
shouldn't you add the settlement contract address just in case it changes?

Reply:
just for completeness.
It's kind of weird solvers gets the `call_data` but not the contract which it is supposed to send to.

Reply:
Added `from` and `to` fields, with a bit of refactoring to make it easier to add additional fields in the future (if needed).